boolean/test_boolean.py

```diff
@@ -819,10 +819,11 @@ def test_simplify_complex_expression_parsed_then_simplified(self):
     def test_parse_invalid_nested_and_should_raise_a_proper_exception(self):
         algebra = BooleanAlgebra()
         expr = '''a (and b)'''
-        try:
+
+        with self.assertRaises(ParseError) as context:
             algebra.parse(expr)
-        except ParseError as pe:
-            assert pe.error_code == PARSE_INVALID_NESTING
+
+        self.assertEqual(context.exception.error_code, PARSE_INVALID_NESTING)
```

Comment:
Good catch as it was missing a self.fail() in the previous form.
It would be more explicit to have this part of the context block, would it?